src/main/java/org/openrewrite/staticanalysis/EqualsAvoidsNullVisitor.java

```diff
@@ -17,63 +17,111 @@
 
 import lombok.EqualsAndHashCode;
 import lombok.Value;
-import org.jspecify.annotations.Nullable;
 import org.openrewrite.Tree;
 import org.openrewrite.java.JavaVisitor;
 import org.openrewrite.java.MethodMatcher;
 import org.openrewrite.java.style.EqualsAvoidsNullStyle;
-import org.openrewrite.java.tree.*;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.Space;
 import org.openrewrite.marker.Markers;
 
 import static java.util.Collections.singletonList;
+import static java.util.Objects.nonNull;
+import static java.util.Objects.requireNonNull;
+import static org.openrewrite.java.tree.JLeftPadded.build;
+import static org.openrewrite.java.tree.Space.SINGLE_SPACE;
 
+/**
+ * A visitor that identifies and addresses potential issues related to
+ * the use of {@code equals} methods in Java, particularly to avoid
+ * null pointer exceptions when comparing strings.
+ * <p>
+ * This visitor looks for method invocations of {@code equals},
+ * {@code equalsIgnoreCase}, {@code compareTo}, and {@code contentEquals},
+ * and performs optimizations to ensure null checks are correctly applied.
+ * <p>
+ * For more details, refer to the PMD best practices:
+ * <a href="https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#LiteralsFirstInComparisons">Literals First in Comparisons</a>
+ *
+ * @param <P> The type of the parent context used for visiting the AST.
+ */
 @Value
 @EqualsAndHashCode(callSuper = false)
 public class EqualsAvoidsNullVisitor<P> extends JavaVisitor<P> {
```

Comment:
Perhaps LiteralsFirstInComparisonsVisitor would be a better fit, as it's no longer just about equality; both equality and comparison are now merely implementation details.
```suggestion
public class LiteralsFirstInComparisonsVisitor<P> extends JavaVisitor<P> {
```

Reply:
I'd hold off on renaming this just yet. While the visitor only appears to be used in this module, the recipe name is well established as part of [our Common static analysis issues recipe](https://docs.openrewrite.org/recipes/staticanalysis/commonstaticanalysis), which folks have created custom copies off that might break if we similarly rename the recipe.

Reply:
Yes thanks. Its just an idea and of course would be a new PR topic.